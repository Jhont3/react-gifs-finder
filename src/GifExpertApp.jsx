import { useState } from "react";
// import { AddCategory } from "./components/AddCategory";
// import { GifGrid } from "./components/GifGrid";
import { AddCategory, GifGrid } from './components'

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['One Punch']);
    // console.log(categories);

    const onAddCategory = ( newCategory ) => {
        // console.log( newCategory );

        if ( categories.includes (newCategory)) return;
        // falta validar si ingresa mayusculas y minusculas lo mismo

        setCategories([ newCategory, ...categories ]);
    };

  return (
    <>
        {/* titulo */}
        
        <h1>Gif's Finder!!!</h1>

        {/* input */}

        <AddCategory 
        // setCategories={ setCategories }
            onNewCategory= { value => onAddCategory( value )}
        />

        {/* listado */}   

        { categories.map( category => (
            <GifGrid 
                key={ category }
                category={ category }
            />
        )) 
        }
    </>
  )
}
