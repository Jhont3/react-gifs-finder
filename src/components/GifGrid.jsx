// import { useEffect, useState } from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GitItem } from "./GitItem";

export const GifGrid = ({ category }) => {
 
    const { images, isLoading } = useFetchGifs( category );


  return (
    <>
      <h3>{category}</h3>
      {
        isLoading && ( <h2>Loading...</h2> )
      }

      <div className="card-grid">
        {
            images.map(( image ) => (
            <GitItem 
                key={ image.id }
                { ...image }
            />
        ))}
      </div>

    </>
  );
};
