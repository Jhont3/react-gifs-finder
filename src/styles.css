* {
    font-family: Helvetica, Arial, sans-serif;
    background-color: rgba(255, 255, 255, 0.941);
    box-sizing: border-box;
    margin: 0px;
  }
  
  body {
    padding: 2% 5% 5%;
    width: 100vw;
  }
  
  input {
    background-color: #fff;
    border-radius: 18px;
    color: #000;
    font-size: 1.2rem;
    outline: none;
    padding: 10px 15px;
    width: 95%;
    margin: 20px 0px;
  }

  h1{
    font-size: 3rem;
    text-align: center;
    margin: 20px 0px;
    box-shadow: rgba(0, 0, 0, 0.17) 0px -23px 25px 0px inset,  rgba(0, 0, 0, 0.1) 0px -79px 40px 0px inset, rgba(0, 0, 0, 0.06) 0px 2px 1px, rgba(0, 0, 0, 0.09) 0px 4px 2px;
    border-radius: 25px;
  }
  
  h2 {
    font-size: 1.5rem;
  }
  
  h3 {
    font-size: 2.5rem;
    margin-bottom: 5px;
   
  }

  .card-grid{
    display: grid;
    min-height: 100vh;
    grid-template-columns: repeat(auto-fill, minmax(300px, 1fr));    
    grid-gap: 1rem 2rem;
    align-items: center;
    
  }

  .card{
    padding: 2%;
    border-radius: 10px;
    box-shadow: rgba(50, 50, 93, 0.25) 0px 6px 12px -2px, rgba(0, 0, 0, 0.3) 0px 3px 7px -3px;
    background-color: white;
    
  }

  .card img{
    width: 100%;
  }

  .card p{
    padding: 5px;
    text-align: center;
  }
