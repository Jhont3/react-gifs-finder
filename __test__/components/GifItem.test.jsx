import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components";


describe('Tarea', () => { 
    const title = 'Mi prueba';
    const url = 'https://a423432.com/'

    test('Match con Snapshot', () => { 
        const { container } = render(<GifItem title={ title } url={ url } />);
        expect( container ).toMatchSnapshot();
     });
    

     test('Debe mostrar la url y la imagen indicada', () => { 
        render(<GifItem title={ title } url={ url } />);
        // screen.debug();
        // expect( screen.getByRole('img').src ).toBe( url );
        // expect( screen.getByRole('img').alt ).toBe( title );
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( title );
    });

    test('Debe mostrar el titulo en el componente', () => { 
        render(<GifItem title={ title } url={ url } />);
        expect( screen.getByText( title )).toBeTruthy();
     });


 });


