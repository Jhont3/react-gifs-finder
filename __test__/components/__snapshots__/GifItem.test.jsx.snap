// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tarea Match con Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Mi prueba"
      src="https://a423432.com/"
    />
    <p>
       
      Mi prueba
       
    </p>
  </div>
</div>
`;
